ARCs/arc-draft_extended_desc.md

```diff
@@ -0,0 +1,160 @@
+---
+arc: <to be assigned>
+title: Extended App Description
+description: Adds more information to the ARC4 JSON description
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-11-14
+requires: 4
+---
+
+
+## Abstract
+This ARC takes the existing JSON description of a contract as described in ARC4 and adds more fields for the purpose of client interaction
+
+## Motivation
+The description provided by ARC4 is missing a lot of critical information that clients should know when interaction with an app.
+
+## Specification
+```ts
+/** Mapping of named structs to the ABI type of their fields */
+interface StructFields {
+  [name: string]: string | StructFields;
+}
+
+/** Describes a single key in app storage */
+interface StorageKey {
```

Comment:
This should have a `description` field too.

Also, is `StorageKey` the right name? Should it be `StorageItem`?

Reply:
I use Key vs Map to differentiate between a single key-value pair and a mapping of key-value pairs. I'm open to other names but for what its worth this is the naming I use in TEALScript

Reply:
Added `desc` in a18a968a16f3bb272c5da8d8e8d90f18eddb7ab8. I went with `desc` because that's what's already used by ARC4 and figured it would be kinda weird to have `desc` some places and `description` in others